src/ServiceInsight/ServiceControl/DefaultServiceControl.cs

```diff
@@ -88,38 +90,37 @@ public Uri CreateServiceInsightUri(StoredMessage message)
 
         public SagaData GetSagaById(Guid sagaId) => GetModel<SagaData>(new RestRequestWithCache(string.Format(SagaEndpoint, sagaId), RestRequestWithCache.CacheStyle.IfNotModified)) ?? new SagaData();
 
-        public PagedResult<StoredMessage> Search(string searchQuery, int pageIndex = 1, string orderBy = null, bool ascending = false)
+        public PagedResult<StoredMessage> GetAuditMessages(Endpoint endpoint, string searchQuery = null, string orderBy = null, bool ascending = false)
         {
-            var request = CreateMessagesRequest();
+            var request = CreateMessagesRequest(endpoint?.Name);
 
             AppendSystemMessages(request);
             AppendSearchQuery(request, searchQuery);
-            AppendPaging(request, pageIndex);
             AppendOrdering(request, orderBy, ascending);
+            AppendPaging(request, DefaultPageSize);
 
             var result = GetPagedResult<StoredMessage>(request);
             if (result != null)
             {
-                result.CurrentPage = pageIndex;
+                result.CurrentPage = 1;
             }
             return result;
         }
 
-        public PagedResult<StoredMessage> GetAuditMessages(Endpoint endpoint, string searchQuery = null, int pageIndex = 1, string orderBy = null, bool ascending = false)
+        public PagedResult<StoredMessage> GetAuditMessages(string link)
         {
-            var request = CreateMessagesRequest(endpoint.Name);
-
-            AppendSystemMessages(request);
-            AppendSearchQuery(request, searchQuery);
-            AppendPaging(request, pageIndex);
-            AppendOrdering(request, orderBy, ascending);
-
-            var result = GetPagedResult<StoredMessage>(request);
-            if (result != null)
+            if (link.StartsWith("http"))
```

Comment:
There are two things that depend on the result of this comparison. I can replace the `if` with `?` but I would end up with two `?` operators.